shell/browser/browser_win.cc

```diff
@@ -180,6 +181,82 @@ bool FormatCommandLineString(base::string16* exe,
+// iterates over all the entries in a windows registry path and returns
+// a list of launchItem with matching paths to our application.
+// if a launchItem with a matching path also has a matching entry within the
+// startup_approved_key_path, set executable_launch_at_login to be `true`
+std::vector<Browser::LaunchItem> GetLoginItemSettingsHelper(
+    base::win::RegistryValueIterator* it,
+    boolean* executable_launch_at_login,
+    base::string16 scope,
+    const Browser::LoginItemSettings& options) {
+  std::vector<Browser::LaunchItem> launch_items;
+
+  while (it->Valid()) {
+    base::string16 exe = options.path;
+    if (FormatCommandLineString(&exe, options.args)) {
+      // add launch item to vector if it has a matching path (case-insensitive)
+      if ((base::CompareCaseInsensitiveASCII(it->Value(), exe.c_str())) == 0) {
+        Browser::LaunchItem launch_item;
+        base::string16 launch_path = options.path;
+        if (!(launch_path.size() > 0)) {
+          GetProcessExecPath(&launch_path);
+        }
+        launch_item.name = it->Name();
+        launch_item.path = launch_path;
+        launch_item.args = options.args;
+        launch_item.scope = scope;
+        launch_item.enabled = true;
+
+        // attempt to update launch_item.enabled if there is a matching key
+        // value entry in the StartupApproved registry
+        HKEY hkey;
+        // StartupApproved registry path
+        LPCTSTR path = TEXT(
+            "SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Explorer\\StartupApp"
+            "roved\\Run");
+        LONG res;
+        if (scope == L"user") {
+          res = RegOpenKeyEx(HKEY_CURRENT_USER, path, 0,
+                             KEY_QUERY_VALUE | KEY_WOW64_64KEY, &hkey);
+        } else {
+          res = RegOpenKeyEx(HKEY_LOCAL_MACHINE, path, 0,
+                             KEY_QUERY_VALUE | KEY_WOW64_64KEY, &hkey);
+        }
+        if (res == ERROR_SUCCESS) {
+          DWORD type, size;
+          wchar_t startup_binary[12];
+          LONG result =
+              RegQueryValueEx(hkey, it->Name(), nullptr, &type,
+                              reinterpret_cast<BYTE*>(&startup_binary),
+                              &(size = sizeof(startup_binary)));
+          if (result == ERROR_SUCCESS) {
+            if (type == REG_BINARY) {
+              // any other binary other than this indicates that the program is
+              // not set to launch at login
+              wchar_t binary_accepted[12] = {0x00, 0x00, 0x00, 0x00,
+                                             0x00, 0x00, 0x00, 0x00,
+                                             0x00, 0x00, 0x00, 0x00};
+              std::string reg_binary(reinterpret_cast<char*>(binary_accepted));
+              std::string reg_startup_binary(
+                  reinterpret_cast<char*>(startup_binary));
+              bool enabled = reg_binary == reg_startup_binary;
+              launch_item.enabled = enabled;
```

Comment:
no need for an extra variable.
```suggestion
              launch_item.enabled = reg_binary == reg_startup_binary;;
```